### DISCUSSION TITLE
Email Editor

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2826] [Office hours] [answered]
Hello! I signed up for office hours this Thursday, but wanted to start gathering context early if possible.

The MC team is working on delivering an email editor for Engage by EOY. The loose reqs are that it matches the current 3rd party's features ([Unlayer](https://www.figma.com/file/AThoZNMoCXViDddWDbKb0d/Email-Editor?node-id=82%3A2019)) in addition to filling in some gaps, plus designing the entire experience in Paste. This editor would ideally scale to cover the needs of MC, and other Twilio products down the line.

Drag & drop template template building is a core function of the editor, and I'm wondering if there's anything I can leverage in Paste to build this. I know the team is currently working on a Dropzone component - would that make sense as a starting point for the drag & drop builder component?

Looking forward to discussing in more detail this Thurs!

**Comment by SiTaggart** [comment]
Hi @ceyerkes 

Unfortunately, the Dropzone Component got prioritized in exchange for a File uploader. You are correct, in that the dropzone component was intended to be useful for this use case, as it will closely align with something similar in [Lightning Design System](https://www.lightningdesignsystem.com/components/drop-zone/). You can probably use something like this to form the basis for drag and drop-able sections and assets inside the email editor. Happy to work with you to flesh out some requirements on this component and that form the basis of a contribution to the system, probably in Q1 at this stage.

We're also adding support for a Rich Text Editor Library called [Lexical](https://lexical.dev/) which we're hoping will form the basis of every Rich Text Editor at Twilio. The idea being we expose the "Lego" and product teams compose them together to create the product experiences they need.

From the office hours session we had, it looks like a lot of the rest might be covered by Paste out of the box, but without super detailed requirements it's hard to tell. 

Looking forward to seeing the output of this.

**Comment by ceyerkes** [comment]
Hi @SiTaggart - I'd love your help in fleshing out requirements for the dropzone component, and eventually contribute back to the system. Let me know what makes sense to kick this off (paste office hours, another dedicated meeting, etc). 

Does it make sense for me to start thinking about how Lexical would fit in to this first version of the editor, or is this further down the road/something we can incorporate later on?

Thanks!

**Reply by SiTaggart** [reply]
Office Hours is probably the best place to start. Ideally in the next two weeks before I go on Recharge.

Lexical is going into the system as a library this week. I think now would be a perfect time to start thinking about how the team can leverage it. You'll need to work with the engineering team pretty closely to make sure it meets the needs and functionality of the email editor, but I'm hopeful it will.

**Reply by ceyerkes** [reply]
Great! I signed up for next week - let me know if you think I should book the whole session, or if 20 mins will be sufficient. I also invited the eng lead on this project, @cfree.

**Comment by gloriliale** [comment, marked as answer]
Hi @ceyerkes! Thanks for chatting with us at Office Hours! [Here is a recording](https://www.loom.com/share/e5ef8cd899d74caea54b03f522fd5b18) of our session (password is in our help channel). Let us know how things progress and how we can continue being of help!
